### DISCUSSION TITLE
A Discussion on Workbench Selectors or the future of ComboBox as a Workbench Selector

### DISCUSSION THREAD

**Discussion by Reqrefusion** [FreeCAD/FreeCAD#13818] [Discussions that will turn in to issues but need refining]
With the addition of TabBar to the core, we now have two Workbench selectors. Of course, #12270 that added the TabBar did not define this feature as default. This would not be very appropriate for a feature added for the first time. But the sun is rising again and we must make new choices. I would like to offer some options in this direction. And yes, I am definitely not impartial.

**Complete removal of ComboBox**

It's debatable how appropriate ComboBox is for a selector, and I call it unsuitable, as the whole world says. No commonly used program uses ComboBox in such a function. First, it increases the number of clicks to get somewhere and causes confusion by displaying too many things that were not on the screen before. Therefore, it would be best to say goodbye to 1.0 completely. Even the parties in the relevant PR who argued that TabBar should not be added to the core stated that they were using TabBar as an add-on. So I don't think there will be any opposition on this issue, especially when TabBar becomes more mature.

**TabBar being Default**

TabBar may not be mature enough right now, but it will be in the future and should definitely be the default. By choosing ComboBox in this election, we are making happy those who want to get nostalgic and reminisce about the old days. However, especially for newcomers, seeing a familiar interface will cause them to become more connected.

**Asking at Start Which Workbench Selector to Use**

In addition to the questions asked at the start, you must also ask which Workbench Selector you want to use if you want to keep two workbench selectors. This is not a choice, it is a must. No matter which default is chosen, the user must be asked about this at Start.

**Continuing the same old system**

Should FreeCad really enter a new era with ruins on it? If it were me, I would say no to this.

**Comment by pierreporte** [comment]
Thanks for your new UI/UX proposal. Would you join the Design Working Group on Discord? Here it is: https://discord.gg/aS4bNnPv

**Reply by Reqrefusion** [reply]
Thank you for asking. Looks like I'll have to move what I wrote here over there.

**Reply by pierreporte** [reply]
Feel free to just explore and have a look first to see if you want to work with us. You’ll be welcome especially if you have experience in UI/UX.

**Reply by Reqrefusion** [reply]
Thank you very much. Yes, there are some, but mostly web related things.

**Comment by pierreporte** [comment]
Like you, I think that the TabBar selector is better in terms of accessibility and ease of use, and a good step towards a modern interface (as you explained in #13775).

The combo box selector could be used as a workspace selector similar to Fusion 360. There would be a mechanical design workspace, a BIM workspace, a simulation workspace, and more if needed. Workspace change is less common that workbench change so I think it is acceptable.

**Reply by Reqrefusion** [reply]
It's a nice idea, but it definitely creates a mess. The Workspace idea is a very good idea regarding the grouping of tools. You say that the Workbench replacement is not used much, but I still do not want to see the ComboBox. So what is your opinion about the current situation? Which should be the default?

**Reply by slyblood** [reply]
I'm a new user to FreeCAD and i've been learning FC off of the weekly builds to see the latest changes and fixes with the major problems with the program(which thanks to all of you for the hard work).

I believe the TabBar is more intuitive to a new user than the ComboBox, however I don't find the WorkBench concept that intuitive for a new user at all and believe it needs a rethink. I believe a lot of the stuff could be consolidated into just a few different "WorkBenches" and instead of thinking of the workbenches as they are now as basically different groups of tools but rather thinking of them in the context of Modules to do actually different things(i.e. 2D Drafting & Architecture, 3D Assembly and Design and Testing(FEM, etc).

This is just the same idea that Blender proposed because the project has gotten so large that people really just want to trim the program down to do just what they want to do for instance video editing or whatever. FreeCAD has a lot of stuff which is great to some degree but also a lot to through at a new user. I believe when the user first starts the program, the start page should ask what they are going to use FreeCAD for and then only install those "workbenches". But they should only be a few to choose from and the workflows within those workbenches and tool selection more context driven and refined.

If I select a face on a part or body or what have you, and I right click on it I should get all of the related tools in that right click menu that able to be used with the selection(i.e. extrude or pad). I think this is what blocks a lot of people from just watching other people try to use FreeCAD for the first time on Youtube and versus other programs because the other CAD programs "FEEL" more simple, like I was left wondering why the mesh workbench even exists when(to me anyway) it seems those tools could just be a simple submenu in a right click box. I was just left wondering why I need to select a whole other workbench just for a few tools.

Also just from a tech debt perspective there should only be one GOOD way to do something, don't let FreeCAD turn into C++. :D

Those are just my thoughts.

**Reply by Reqrefusion** [reply]
@slyblood Thank you for joining the discussion. I'd like to receive your praise, but I'm just hanging around here. Reconsidering the Workbench concept completely is a philosophical problem beyond the technical problem, something that requires mastery of the entire product and can only be done by someone like the product owner. More concrete mergers can be proposed to help with this. Concrete thought is closer to the result.
